public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -468,6 +473,13 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
             // Put ids in array for query to get all results at once
             List<String> qlist = new ArrayList();
             for (QuestionConcept q : questions) {
+                if (q.getSubQuestionCount() > 0) {
+                    List<QuestionConcept> subQuestions = questionConceptDao.findSubSurveyQuestions(surveyConceptId, q.getConceptId());
+                    List<Long> subQuestionConceptIds = subQuestions.stream().map(QuestionConcept::getConceptId).collect(Collectors.toList());
```

Comment:
It would probably be cleaner to merge these three things into a single stream operation:

```
List<String> subQuestionIds = questionConceptDao.findSubSurveyQuestions(surveyConceptId, q.getConceptId()).stream()
  .map(QuestionConcept::getConceptId)
  .map(String::valueOf)
  .collect(...);
```

Reply:
Done. Thanks